### PR TITLE
feat: merge command can now create merge commits (for very simple cases)

### PR DESCRIPTION
**Pull request by wmhilton**
Adds `author`, `commiter`, and `signingKey` args to `merge` which get passed on to `commit` if needed.

## I'm adding a parameter to an existing command:

- [x] add parameter to the function in `src/commands/X.js`
- [x] document the parameter in the JSDoc comment above the function
- [x] add parameter to the TypeScript library definition for X in `src/index.d.ts`
- [x] add a test case in `__tests__/test-X.js` if possible
- [ ] squash merge the PR with commit message "feat: Added 'bar' parameter to X command"

### CODE REVIEW

**Comment by karma-pr-reporter** [comment]
## Test Results for dabbe2a32449f866e8a2ead637eaf0e95dd61b7a:
| Browser | Passed | Skipped | Failed | Time | Disconnected | Error |
|---------|--------|---------|--------|------|--------------|-------|
| HeadlessChrome 0.0.0 (Linux 0.0.0) | 287 | 1 | 0 | 1 min 45.103 secs | false | undefined |
| Firefox 68.0.0 (Ubuntu 0.0.0) | 287 | 1 | 0 | 2 mins 23.551 secs | false | undefined |
| Safari 11.1.2 (Mac OS X 10.13.6) | 287 | 1 | 0 | 4 mins 31.847 secs | false | undefined |
| Chrome Mobile 67.0.3396 (Android 7.1.1) | 287 | 1 | 0 | 4 mins 35.052 secs | false | undefined |
| Edge 17.17134.0 (Windows 10 0.0.0) | 287 | 1 | 0 | 6 mins 8.504 secs | false | undefined |
| Mobile Safari 11.0.0 (iOS 11.2.0) | 287 | 1 | 0 | 7 mins 46.382 secs | false | undefined |

**Comment by karma-pr-reporter** [comment]
## Test Results for 58aa8adaed6e2651fff2050b80d0b3f503af3465:
| Browser | Passed | Skipped | Failed | Time | Disconnected | Error |
|---------|--------|---------|--------|------|--------------|-------|
| HeadlessChrome 0.0.0 (Linux 0.0.0) | 288 | 1 | 0 | 1 min 33.271 secs | false | undefined |
| Firefox 68.0.0 (Ubuntu 0.0.0) | 288 | 1 | 0 | 2 mins 12.521 secs | false | undefined |
| Safari 11.1.2 (Mac OS X 10.13.6) | 288 | 1 | 0 | 4 mins 21.985 secs | false | undefined |
| Chrome Mobile 67.0.3396 (Android 7.1.1) | 288 | 1 | 0 | 4 mins 29.298 secs | false | undefined |
| Mobile Safari 11.0.0 (iOS 11.2.0) | 288 | 1 | 0 | 4 mins 46.183 secs | false | undefined |
| Edge 17.17134.0 (Windows 10 0.0.0) | 288 | 1 | 0 | 5 mins 59.165 secs | false | undefined |

**Comment by karma-pr-reporter** [comment]
## Test Results for 17c48787ff147d05397d5a5cdbadc40607acff6e:
| Browser | Passed | Skipped | Failed | Time | Disconnected | Error |
|---------|--------|---------|--------|------|--------------|-------|
| HeadlessChrome 0.0.0 (Linux 0.0.0) | 289 | 1 | 0 | 1 min 42.315 secs | false | undefined |
| Firefox 68.0.0 (Ubuntu 0.0.0) | 289 | 1 | 0 | 2 mins 20.386 secs | false | undefined |
| Safari 11.1.2 (Mac OS X 10.13.6) | 289 | 1 | 0 | 4 mins 27.4 secs | false | undefined |
| Chrome Mobile 67.0.3396 (Android 7.1.1) | 289 | 1 | 0 | 4 mins 59.49 secs | false | undefined |
| Mobile Safari 11.0.0 (iOS 11.2.0) | 289 | 1 | 0 | 5 mins 16.103 secs | false | undefined |
| Edge 17.17134.0 (Windows 10 0.0.0) | 289 | 1 | 0 | 7 mins 5.895 secs | false | undefined |

**Comment by karma-pr-reporter** [comment]
## Test Results for df11d0c9cb14425a58f4024b56c0303b04c32891:
| Browser | Passed | Skipped | Failed | Time | Disconnected | Error |
|---------|--------|---------|--------|------|--------------|-------|
| HeadlessChrome 0.0.0 (Linux 0.0.0) | 291 | 1 | 0 | 1 min 38.273 secs | false | undefined |
| Firefox 68.0.0 (Ubuntu 0.0.0) | 291 | 1 | 0 | 2 mins 17.856 secs | false | undefined |
| Safari 11.1.2 (Mac OS X 10.13.6) | 291 | 1 | 0 | 4 mins 30.739 secs | false | undefined |
| Chrome Mobile 67.0.3396 (Android 7.1.1) | 291 | 1 | 0 | 4 mins 44.188 secs | false | undefined |
| Mobile Safari 11.0.0 (iOS 11.2.0) | 291 | 1 | 0 | 4 mins 54.913 secs | false | undefined |
| Edge 17.17134.0 (Windows 10 0.0.0) | 291 | 1 | 0 | 6 mins 48.19 secs | false | undefined |

File: src/models/GitError.js [hunk 1]
```diff
@@ -71,7 +71,8 @@ const messages = {
   PluginSchemaViolation: `Schema check failed for "{ plugin }" plugin; missing { method } method.`,
   PluginUnrecognized: `Unrecognized plugin type "{ plugin }"`,
   AmbiguousShortOid: `Found multiple oids matching "{ short }" ({ matches }). Use a longer abbreviation length to disambiguate them.`,
-  ShortOidNotFound: `Could not find an object matching "{ short }".`
+  ShortOidNotFound: `Could not find an object matching "{ short }".`,
+  MergeConflict: `Merge conflicts occurred. See 'data' for details.`
```
Comment (wmhilton) [inline src/models/GitError.js]: um.... do I actually include anything in `data`?

File: src/models/FileSystem.js [hunk 2]
```diff
@@ -143,6 +143,19 @@ export class FileSystem {
...
+   */
+  async rmdir (filepath) {
+    try {
+      await this._rmdir(filepath)
+      return true
+    } catch (err) {
+      return false
+    }
+  }
+
```
Comment (wmhilton) [inline src/models/FileSystem.js]: todo: delete this?

File: src/commands/_diffTree.js [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+          return parent
+        }
+        // // If we're deleting a directory, we need to order that _after_ children.
+        // if (parent && parent.ops.includes('rmdir')) {
+        //   children.push(parent)
+        // // Everything else, do the parent operations first.
+        // } else if (parent) {
+        //   children.unshift(parent)
+        // }
+        // return flat(children)
```
Comment (wmhilton) [inline src/commands/_diffTree.js]: remove commented code

**Comment by karma-pr-reporter** [comment]
## Test Results for 403b2c7c48f41574743e3ed07b11dafb0b77488b:
| Browser | Passed | Skipped | Failed | Time | Disconnected | Error |
|---------|--------|---------|--------|------|--------------|-------|
| HeadlessChrome 0.0.0 (Linux 0.0.0) | 293 | 1 | 0 | 1 min 41.962 secs | false | undefined |
| Firefox 68.0.0 (Ubuntu 0.0.0) | 293 | 1 | 0 | 2 mins 20.158 secs | false | undefined |
| Chrome Mobile 67.0.3396 (Android 7.1.1) | 293 | 1 | 0 | 6 mins 18.044 secs | false | undefined |
| Safari 11.1.2 (Mac OS X 10.13.6) | 292 | 1 | 1 | 6 mins 43.233 secs | false | undefined |
| Edge 17.17134.0 (Windows 10 0.0.0) | 293 | 1 | 0 | 8 mins 8.277 secs | false | undefined |
| Mobile Safari 11.0.0 (iOS 11.2.0) | 249 | 1 | 0 | 9 mins 24.582 secs | true | undefined |

**Comment by karma-pr-reporter** [comment]
## Test Results for 403b2c7c48f41574743e3ed07b11dafb0b77488b:
| Browser | Passed | Skipped | Failed | Time | Disconnected | Error |
|---------|--------|---------|--------|------|--------------|-------|
| Safari 11.1.2 (Mac OS X 10.13.6) | 293 | 1 | 0 | 4 mins 29.625 secs | false | undefined |
| Mobile Safari 11.0.0 (iOS 11.2.0) | 293 | 1 | 0 | 4 mins 57.46 secs | false | undefined |

**Comment by karma-pr-reporter** [comment]
## Test Results for 8249ef6e86e68a34e8c798569c2b83b929f4f9cb:
| Browser | Passed | Skipped | Failed | Time | Disconnected | Error |
|---------|--------|---------|--------|------|--------------|-------|
| HeadlessChrome 0.0.0 (Linux 0.0.0) | 296 | 1 | 0 | 1 min 34.918 secs | false | undefined |
| Firefox 68.0.0 (Ubuntu 0.0.0) | 296 | 1 | 0 | 2 mins 11.031 secs | false | undefined |
| Chrome Mobile 67.0.3396 (Android 7.1.1) | 150 | 0 | 0 | 2 mins 57.172 secs | false | undefined |
| Safari 11.1.2 (Mac OS X 10.13.6) | 296 | 1 | 0 | 4 mins 58.3 secs | false | undefined |
| Mobile Safari 11.0.0 (iOS 11.2.0) | 296 | 1 | 0 | 5 mins 7.771 secs | false | undefined |
| Edge 17.17134.0 (Windows 10 0.0.0) | 285 | 1 | 0 | 5 mins 55.138 secs | false | undefined |

**Pull request merged**